File: Swift/LeetCode/LeetCode2574.swift
// https://leetcode.com/problems/left-and-right-sum-differences

class Solution {
    func leftRightDifference(_ nums: [Int]) -> [Int] {
        var answer = [Int](repeating: 0, count: nums.count)
        let totalSum = nums.reduce(0) { $0 + $1 }
        var leftSum = 0
        for idx in 0..<nums.count {
            let rightSum = totalSum - leftSum - nums[idx]
            answer[idx] = abs(leftSum - rightSum)
            leftSum += nums[idx]
        }
        return answer
    }
}

File: C++/LeetCode/LeetCode2574.cpp
// https://leetcode.com/problems/left-and-right-sum-differences

#include <vector>
#include <numeric>
using namespace std;

class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        const int totalSum = accumulate(nums.begin(), nums.end(), 0);
        vector<int> answer(nums.size());
        int leftSum = 0;
        for (int idx = 0; idx < nums.size(); idx++) {
            const int rightSum = totalSum - leftSum - nums[idx];
            answer[idx] = abs(leftSum - rightSum);
            leftSum += nums[idx];
        }
        return answer;
    }
};